UDPClientGUI/MainWindow.xaml.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Documents;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.Windows.Navigation;
+using System.Windows.Shapes;
+
+namespace UDPCommGUI
+{
+    /// <summary>
+    /// Interaction logic for MainWindow.xaml
+    /// </summary>
+    public partial class MainWindow : Window
+    {
+
+        MyUDPClient client = new MyUDPClient();
+
+        public MainWindow()
+        {
+            InitializeComponent();
+        }
+
+        private void IPChanged(object sender, TextChangedEventArgs e)
+        {
+            client.ConnectionIP = IPbox.Text;
+        }
+
+        private void PortChanged(object sender, TextChangedEventArgs e)
+        {
+            try
```

Comment:
Why is there a try/finally? You could just use int.TryParse

Note that `Int32` is the same as `int` but you'd rarely see the first one.

Have a read of `ref`, `in` and `out` while you're at it

Reply:
Changed as suggested.